packages/framework/ai-collab/src/test/explicit-strategy/agentEditingReducer.spec.ts

```diff
@@ -380,84 +381,207 @@ describe("applyAgentEdit", () => {
 		});
 	});
 
-	it("modify edits", () => {
-		const tree = factory.create(
+	describe("modify edits", () => {
+		let tree = factory.create(
 			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
 			"tree",
 		);
-		const config = new TreeViewConfiguration({ schema: RootObjectPolymorphic });
-		const view = tree.viewWith(config);
-		const simpleSchema = getSimpleSchema(view.schema);
+		let config = new TreeViewConfiguration({ schema: RootObjectPolymorphic });
+		let view = tree.viewWith(config);
+		let simpleSchema = getSimpleSchema(view.schema);
 
-		view.initialize({
-			str: "testStr",
-			vectors: [new Vector({ x: 1, y: 2, z: 3 })],
-			bools: [true],
+		// Reinitialize our test tree to the same initial state.
+		beforeEach(() => {
+			tree = factory.create(
+				new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+				"tree",
+			);
+			config = new TreeViewConfiguration({ schema: RootObjectPolymorphic });
+			view = tree.viewWith(config);
+			simpleSchema = getSimpleSchema(view.schema);
+
+			view.initialize({
+				str: "testStr",
+				vectors: [new Vector({ x: 1, y: 2, z: 3 })],
+				bools: [true],
+			});
+
+			idGenerator.assignIds(view.root);
 		});
 
-		idGenerator.assignIds(view.root);
-		const vectorId = idGenerator.getId(view.root as TreeNode) ?? fail("ID expected.");
+		it("replace an array node of objects", () => {
+			const vectorArrayId = idGenerator.getId(view.root as TreeNode) ?? fail("ID expected.");
 
-		const modifyEdit: TreeEdit = {
-			explanation: "Modify a vector",
-			type: "modify",
-			target: { target: vectorId },
-			field: "vectors",
-			modification: [
-				{ [typeField]: Vector.identifier, x: 2, y: 3, z: 4 },
-				{ [typeField]: Vector2.identifier, x2: 3, y2: 4, z2: 5 },
-			],
-		};
-		applyAgentEdit(modifyEdit, idGenerator, simpleSchema.definitions);
+			const modifyEdit: TreeEdit = {
+				explanation: "Replace an array of vectors",
+				type: "modify",
+				target: { target: vectorArrayId },
+				field: "vectors",
+				modification: [
+					{ [typeField]: Vector.identifier, x: 2, y: 3, z: 4 },
+					{ [typeField]: Vector2.identifier, x2: 3, y2: 4, z2: 5 },
+				],
+			};
+			applyAgentEdit(modifyEdit, idGenerator, simpleSchema.definitions);
 
-		const modifyEdit2: TreeEdit = {
-			explanation: "Modify a vector",
-			type: "modify",
-			target: { target: vectorId },
-			field: "bools",
-			modification: [false],
-		};
-		applyAgentEdit(modifyEdit2, idGenerator, simpleSchema.definitions);
+			const expected = {
+				"str": "testStr",
+				"vectors": [
+					{
+						"id": (view.root.vectors[0] as Vector).id,
```

Comment:
I see `id` is expected for `vectors` but not for primitive values such as `string` or `bool`. Is this expected? Does `idGenerator()` generate and embed `id` to non-primitive nodes?

Reply:
Our schema explicitly adds an `id` field to `Vector` (lines 40-52), that's why it only shows up there. It's not that non-primitives get ids by default, it's only those whose schema says they have one.